optimum/habana/transformers/generation/utils.py

```diff
@@ -542,6 +542,9 @@ def generate(
             generation_config.ignore_eos = kwargs.get("ignore_eos", lazy_mode)
         generation_config.validate()
         model_kwargs = generation_config.update(**kwargs)  # All unused kwargs must be model kwargs
+        if self.config.model_type == "falcon" and "token_type_ids" in kwargs.keys():
```

Comment:
Please test if this logic does not get activated for other models using text-generation.

Reply:
tested with other models and made sure it's not used for them